Tribes/src/app/header.service.ts

```diff
@@ -28,4 +29,9 @@ export class HeaderService {
       catchError(this.errorHandlingService.handleError)
     );
   }
+
+  logoutUser() {
+    this.router.navigateByUrl('/login');
+    localStorage.removeItem('TOKEN');
```

Comment:
A little more than a gut feeling I have about this: 
When you navigate, a lot of things are happening. In plain JS it's a native call, which might trigger other synchronousl events, that may fail or throw exception. In that case, the remove token line might not get called. I would remove the token first, then navigate to login in this case.